app/controllers/posts_controller.rb

```diff
@@ -1,14 +1,15 @@
 class PostsController < ApplicationController
-  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :set_post, only: [:show, :edit, :update, :destroy, :favorite, :unfavorite]
+  before_action :block_unauthorized_changes, only: [:edit, :update, :destroy]
   before_filter :authenticate_user!
   # GET /posts
-  # GET /posts.json
   def index
-    @posts = Post.all.includes(:user)
+    # @posts = Post.all.includes(:user)
+    @posts = Post.where(user_id: current_user.following.map(&:id)).includes(:user)
```

Comment:
This is good. I just want to point one thing out to make sure you realize it.

There are two DB queries happening here. The first is from the `current_user.following.map(&:id)`, and the second is from the `Posts.where`.

If you had done `current_user.following.map(&:tweets)`, that would be doing a query for ever user you're following, which is bad if you're following 10,000 users.

Good job!